### PR TITLE
Jinghan/implement offline method `Snapshot`

### PR DESCRIPTION
**Pull request by jinghancc**
This PR implements offline method `Snapshot`, which aggregates the previous snapshot and current cdc table, generates a new snapshot table.

related to #805

### CODE REVIEW

File: internal/database/offline/sqlutil/snapshot.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+		currSnapshotTableName = fmt.Sprintf("%s.%s", *dbOpt.DatasetID, currSnapshotTableName)
+		prevSnapshotTableName = fmt.Sprintf("%s.%s", *dbOpt.DatasetID, prevSnapshotTableName)
+		currCdcTableName = fmt.Sprintf("%s.%s", *dbOpt.DatasetID, currCdcTableName)
+	}
+
+	schema, err := dbutil.BuildCreateSchema(currSnapshotTableName, opt.Group.Entity, opt.Features, dbOpt.Backend)
+	if err != nil {
+		return err
+	}
+	fmt.Println(schema)
```
Comment (YikSanChan) [inline internal/database/offline/sqlutil/snapshot.go:76]: rm

File: internal/database/offline/sqlutil/snapshot.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		FeatureNames:          opt.Features.Names(),
+		PrevSnapshotTableName: prevSnapshotTableName,
+		CurrSnapshotTableName: currSnapshotTableName,
+		CurrCdcTableName:      currCdcTableName,
+		Backend:               dbOpt.Backend,
+	})
+	if err != nil {
+		return err
+	}
+	fmt.Println(query)
```
Comment (YikSanChan) [inline internal/database/offline/sqlutil/snapshot.go:92]: rm

**Review by YikSanChan — APPROVED** [review]
lgtm

**Review by wfxr — APPROVED** [review]
LGTM!

**Review by lianxmfor — APPROVED** [review]
lgtm

**Pull request merged**